### PR TITLE
TINY-9754: now fontsizeinput revert to old value instead of setting a default value

### PR DESCRIPTION
**Pull request by lorenzo-pomili**
Related Ticket: TINY-9754

Description of Changes:

possible solutions to avoid that inserting a value with an invalid unit the value will be set to `0pt` (or any other default unit)

1. a mechanism that reverses it to the previous value.
2. check what value the user is trying to insert and add the default unit

I implemented the first one because the second one has these 2 problems:
- value: `abc` -> no number so back to the previous default.
- from `16pt` to `1emm` -> probably here the intention of the user would have been to go to `1em` but we cannot know that so we would convert it to `1pt` which is far away from the user intention.

**Other Change:**

- put a `defautlValue` different from `0` to cover other possible edge cases with a default that makes "more sense"

Pre-checks:
* [x] Changelog entry added
* [x] Tests have been added (if applicable)
* [x] Branch prefixed with `feature/`, `hotfix/` or `spike/`

Review:
* [x] Milestone set
* [x] ~Docs ticket created (if applicable)~

GitHub issues (if applicable):

### CODE REVIEW

File: modules/tinymce/src/themes/silver/test/ts/browser/editor/toolbar/NumberInputTest.ts [hunk 1]
```diff
@@ -319,4 +319,22 @@ describe('browser.tinymce.themes.silver.throbber.NumberInputTest', () => {
...
+    TinyUiActions.clickOnToolbar(editor, '.tox-number-input input');
+
+    const input = TinyUiActions.clickOnToolbar<HTMLInputElement>(editor, '.tox-number-input input');
+    UiControls.setValue(input, '15invalid_unit');
+    const root = SugarShadowDom.getRootNode(TinyDom.targetElement(editor));
+    FocusTools.setFocus(root, '.tox-number-input input');
+    await FocusTools.pTryOnSelector('Focus should be on input', root, '.tox-number-input input');
+    TinyUiActions.keystroke(editor, Keys.enter());
+
+    TinyAssertions.assertContent(editor, `<p style="font-size: ${originalFontSize};">a<span style="font-size: ${originalFontSize};">b</span>c</p>`);
```
Comment (vpyshnenko) [inline modules/tinymce/src/themes/silver/test/ts/browser/editor/toolbar/NumberInputTest.ts:338]: Consider checking for the fontsizeinput value is reverted

File: modules/tinymce/CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## Unreleased
 
+### Improved
+- Now triyng to set a font size with an invalid unit in `fontsizeinput` cause to get back to the original value. #TINY-9754
+
 ### Fixed
```
Comment (vpyshnenko) [inline modules/tinymce/CHANGELOG.md:12]:
```suggestion
### Fixed
- Setting an invalid unit in the fontsizeinput would not have any effect on the font size and revert it back to the original value. #TINY-9754
```

File: modules/tinymce/src/themes/silver/main/ts/ui/core/complex/BespokeNumberInput.ts [hunk 3]
```diff
@@ -17,8 +17,11 @@ interface BespokeSelectApi {
   readonly getComponent: () => AlloyComponent;
 }
 
+const defautlValue = 16;
```
Comment (tiny-ben-tran) [inline modules/tinymce/src/themes/silver/main/ts/ui/core/complex/BespokeNumberInput.ts:20]:
```suggestion
const defaultValue = 16;
```

File: modules/tinymce/src/themes/silver/test/ts/browser/editor/toolbar/NumberInputTest.ts [hunk 4]
```diff
@@ -319,4 +319,22 @@ describe('browser.tinymce.themes.silver.throbber.NumberInputTest', () => {
     TinyUiActions.keystroke(editor, Keys.enter());
     assert.isTrue(editor.hasFocus(), 'after enter editor should have focus');
   });
+
+  it('TINY-9754: changing unit to an invalid unit should not set the size to 0', async () => {
+    const editor = hook.editor();
+    const originalFontSize = '16px';
```
Comment (tiny-ben-tran) [inline modules/tinymce/src/themes/silver/test/ts/browser/editor/toolbar/NumberInputTest.ts:325]: I would use a different value and unit because 16 is the default value

File: modules/tinymce/src/themes/silver/test/ts/browser/editor/toolbar/NumberInputTest.ts [hunk 5]
```diff
@@ -319,4 +319,22 @@ describe('browser.tinymce.themes.silver.throbber.NumberInputTest', () => {
...
+    editor.setContent(`<p style="font-size: ${originalFontSize};">abc</p>`);
+    TinySelections.setSelection(editor, [ 0, 0 ], 1, [ 0, 0 ], 2);
+
+    TinyUiActions.clickOnToolbar(editor, '.tox-number-input input');
+
+    const input = TinyUiActions.clickOnToolbar<HTMLInputElement>(editor, '.tox-number-input input');
+    UiControls.setValue(input, '15invalid_unit');
```
Comment (tiny-ben-tran) [inline modules/tinymce/src/themes/silver/test/ts/browser/editor/toolbar/NumberInputTest.ts:332]: Check if the fontsizeinput has a correct value-unit text

**Review by tiny-ben-tran — APPROVED** [review]

File: modules/tinymce/src/themes/silver/test/ts/browser/editor/toolbar/NumberInputTest.ts (hunk 1 above)
Comment (lorenzo-pomili) [reply modules/tinymce/src/themes/silver/test/ts/browser/editor/toolbar/NumberInputTest.ts:341]: done

File: modules/tinymce/CHANGELOG.md (hunk 2 above)
Comment (lorenzo-pomili) [reply modules/tinymce/CHANGELOG.md]: moved

File: modules/tinymce/src/themes/silver/main/ts/ui/core/complex/BespokeNumberInput.ts (hunk 3 above)
Comment (lorenzo-pomili) [reply modules/tinymce/src/themes/silver/main/ts/ui/core/complex/BespokeNumberInput.ts]: changed

File: modules/tinymce/src/themes/silver/test/ts/browser/editor/toolbar/NumberInputTest.ts (hunk 4 above)
Comment (lorenzo-pomili) [reply modules/tinymce/src/themes/silver/test/ts/browser/editor/toolbar/NumberInputTest.ts]: :+1: , changed

File: modules/tinymce/src/themes/silver/test/ts/browser/editor/toolbar/NumberInputTest.ts (hunk 5 above)
Comment (lorenzo-pomili) [reply modules/tinymce/src/themes/silver/test/ts/browser/editor/toolbar/NumberInputTest.ts]: done

File: modules/tinymce/src/themes/silver/main/ts/ui/core/complex/BespokeNumberInput.ts [hunk 6]
```diff
@@ -132,6 +137,7 @@ const createBespokeNumberInput = (editor: Editor, backstage: UiFactoryBackstage,
           ]),
           AddEventsBehaviour.config('input-update-display-text', [
             AlloyEvents.run<UpdateMenuTextEvent>(updateMenuText, (comp, se) => {
+              oldValue = Optional.some(se.event.text);
```
[the remark this answers is not in the document]
Comment (lorenzo-pomili) [reply modules/tinymce/src/themes/silver/main/ts/ui/core/complex/BespokeNumberInput.ts:140]: I actually didn't do it because I wasn't sure how to do so, the only way that I found is to use `editor.queryCommandValue('FontSize')`, similar to how it is calculated in [FontSizeBespoke.ts](https://github.com/tinymce/tinymce/blob/develop/modules/tinymce/src/themes/silver/main/ts/ui/core/complex/FontSizeBespoke.ts#L142), and I thought that it would be incorrect since this is `BespokeNumberInput` and it shouldn't be bound to the use of the `font size` (it could be any other kind of number) but then I noticed that we already use the `Dimension.parse` which bind the entire component to the fact that we are using it for the `font size`, now I see these possible ways to solve it:

1. rename the `BespokeNumberInput` in `BespokeFontSizeInput` and use `editor.queryCommandValue('FontSize')`
2. find a way to also remove the `Dimension.parse` and don't use it nor `editor.queryCommandValue('FontSize')`
3. use `editor.queryCommandValue('FontSize')` and when we will need to use this component for a different kind of number, refactor it

File: modules/tinymce/CHANGELOG.md [hunk 7]
```diff
@@ -10,6 +10,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - The `urlinput` dialog component would not open the typeahead dropdown when the input value was reset to an empty string. #TINY-9717
 - Fixed the mouse pointer style from a text cursor to a default arrow pointer when hovering over the tree dialog component items. #TINY-9692
 - Enabled variant of togglable `tox-button` and `tox-button--secondary` now supports `hover`/`active`/`focus`/`disabled` states. #TINY-9713
+- Setting an invalid unit in the `fontsizeinput` would not have any effect on the font size and revert it back to the original value. #TINY-9754
```
[the remark this answers is not in the document]
Comment (lorenzo-pomili) [reply modules/tinymce/CHANGELOG.md]: changed

**Review by TheSpyder — APPROVED** [review]
I hadn't quite realised this field was doing validation. But it makes sense. I'll keep that in mind if/when we ever build a generic validation feature for silver components.

File: modules/tinymce/src/themes/silver/main/ts/ui/core/complex/BespokeNumberInput.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (lorenzo-pomili) [reply modules/tinymce/src/themes/silver/main/ts/ui/core/complex/BespokeNumberInput.ts]: ok, in the end the split the `FontSize` functionality from `BespokeNumberInput`, I move the parse logic from in `FontSizeBespoke.ts`.

So now `validation` and `defaultValue` is demanded to `FontSizeBespoke.ts` and `BespokeNumberInput` require only a `getNewValue` function
